src/connectivity.jl

```diff
@@ -219,31 +219,32 @@ julia> strongly_connected_components(g)
 
 function strongly_connected_components end
 # see https://github.com/mauro3/SimpleTraits.jl/issues/47#issuecomment-327880153 for syntax
-@traitfn function strongly_connected_components(g::AG::IsDirected) where {T<:Integer, AG <: AbstractGraph{T}}
+empty_graph_data(type,g::AG) where {T, AG <: AbstractGraph{T}} = Dict{T,type}()
+empty_graph_data(type,g::AG) where {T<:Integer, AG <: AbstractGraph{T}} = zeros(type,nv(g))
+is_unvisited(data::AbstractVector,v::Integer) = iszero(data[v])
+is_unvisited(data::Dict,v) = !haskey(data,v)
+
+@traitfn function strongly_connected_components(g::AG::IsDirected) where {T, AG <: AbstractGraph{T}}
```

Comment:
It doesn't hurt that you removed the condition that `T` is a subtype of `Integer` but currently the `AbstractGraph` interface does not allow any other `eltype` than some `Integer` (and they are consecutive from 1 to `nv(g)`), so you probably do not need to use a `Dict`.

Reply:
I see. The dict is only ever used as a fallback if T is not an Integer (for example, if someone abused the interface by inheriting from AbstractGraph with T = Symbol), otherwise the code uses vectors by default. The extra lines with the fallbacks are optional.

Reply:
I that case, I would add the `{T <: Integer}` back - T don't think we should handle the case that the interface is incorrectly implemented as it is not clear, where something could go wrong.